### DISCUSSION TITLE
Koan compiled binary

### DISCUSSION THREAD

**Discussion by 1392273211** [cobbler/koan#99] [Q&A]
Koan failed to compile binary. The compiled binary cannot be used. Is there any document on how to compile it into rpm?

**Comment by SchoolGuy** [comment]
Since Koan is Python-based I don't know what binary you are referring to. Sadly we don't have instructions in our ReadTheDocs documentation but you can see examples on how to compile to RPM in GitHub Actions: https://github.com/cobbler/koan/blob/main/.github/workflows/packaging.yml

**Reply by 1392273211** [reply]
Binary is something that can be directly used by placing it in/usr/bin/or compiling it into an rpm package

**Reply by SchoolGuy** [reply]
Albeit this description of a binary is incorrect this is what I described. You can see examples how to compile an RPM in the linked GitHub Action.

**Reply by 1392273211** [reply]
The binary I am referring to is the koan in /usr/local/bin/koan. Is there any corresponding compilation method? Assuming that the binary file koan is compiled on the rock8.6 public network machine, can it be used in the intranet environment?

**Reply by SchoolGuy** [reply]
The file you are referring to is not a binary. This file is the Python entrypoint script that will execute the Python code that is present in the Python site directory of the machine.
